### DISCUSSION TITLE
src for System.Collections.Generic.List<T>

### DISCUSSION THREAD

**Discussion by ActivistInvestor** [dotnet/runtime#98773] [General] [answered]
Looking in [runtime](https://github.com/dotnet/runtime/tree/main)/[src](https://github.com/dotnet/runtime/tree/main/src)/[libraries](https://github.com/dotnet/runtime/tree/main/src/libraries)/[System.Collections](https://github.com/dotnet/runtime/tree/main/src/libraries/System.Collections)/[src](https://github.com/dotnet/runtime/tree/main/src/libraries/System.Collections/src)/[System](https://github.com/dotnet/runtime/tree/main/src/libraries/System.Collections/src/System)/[Collections](https://github.com/dotnet/runtime/tree/main/src/libraries/System.Collections/src/System/Collections)

I do not see list.cs.

The question I'm trying to answer is whether the constructor that takes an `IEnumerable<T>` uses `IIListProvider<T>`.  In .NET 6.0, it does not.

Can someone tell me where I can find it?

Thanks.

**Comment by tannergooding** [comment, marked as answer]
A large number of types exist as part of `System.Private.Corelib` for layering reasons and so do not live in same place as the reference assembly that exposes it publicly.

In general you can use https://source.dot.net to locate types and then click the `Web Access` link to find the specific version of the file that had been indexed. For example, `List<T>` is https://source.dot.net/#System.Private.CoreLib/src/libraries/System.Private.CoreLib/src/System/Collections/Generic/List.cs,cf7f4095e4de7646

The Web Access link at the bottom of the page then goes to https://github.com/dotnet/runtime/blob/8cc7586135856499ea5123bbfb71a5df3fa7f717/src/libraries/System.Private.CoreLib/src/System/Collections/Generic/List.cs

The latest version of this file (which currently happens to be the same) is https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/Collections/Generic/List.cs

**Reply by ActivistInvestor** [reply]
Thanks, I found it.  But I'm puzzled by what I'm seeing. I'll make an issue out of this.
